# SPDX-FileCopyrightText: Huawei Inc.
#
# SPDX-License-Identifier: Apache-2.0

Patch for //base/startup/appspawn_standard repository of OpenHarmony 3.1 codebase.

Prevent buffer overflow error in AppSpawn::SetProcessName() when processName is
longer than longProcNameLen - 1.

Signed-off-by: Esben Haabendal <esben.haabendal@huawei.com>
Upstream-Status: Pending

diff --git a/src/appspawn_server.cpp b/src/appspawn_server.cpp
index 54f66450b81e..707aed8c924e 100644
--- a/src/appspawn_server.cpp
+++ b/src/appspawn_server.cpp
@@ -514,7 +514,7 @@ int32_t AppSpawnServer::SetProcessName(
     }
 
     // set long process name
-    if (strncpy_s(longProcName, len, processName, len) != EOK) {
+    if (strncpy_s(longProcName, longProcNameLen, processName, len) != EOK) {
         HiLog::Error(LABEL, "strncpy_s long name error: %{public}d", errno);
         return -EINVAL;
     }
